{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0daf2864",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting face-recognition\n",
      "  Using cached face_recognition-1.3.0-py2.py3-none-any.whl (15 kB)\n",
      "Requirement already satisfied: Click>=6.0 in c:\\users\\sahil raghuvanshi\\ana\\lib\\site-packages (from face-recognition) (8.0.4)\n",
      "Requirement already satisfied: numpy in c:\\users\\sahil raghuvanshi\\ana\\lib\\site-packages (from face-recognition) (1.21.5)\n",
      "Collecting dlib>=19.7\n",
      "  Using cached dlib-19.24.1.tar.gz (3.2 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Requirement already satisfied: face-recognition-models>=0.3.0 in c:\\users\\sahil raghuvanshi\\ana\\lib\\site-packages (from face-recognition) (0.3.0)\n",
      "Requirement already satisfied: Pillow in c:\\users\\sahil raghuvanshi\\ana\\lib\\site-packages (from face-recognition) (9.2.0)\n",
      "Requirement already satisfied: colorama in c:\\users\\sahil raghuvanshi\\ana\\lib\\site-packages (from Click>=6.0->face-recognition) (0.4.5)\n",
      "Building wheels for collected packages: dlib\n",
      "  Building wheel for dlib (setup.py): started\n",
      "  Building wheel for dlib (setup.py): finished with status 'error'\n",
      "  Running setup.py clean for dlib\n",
      "Failed to build dlib\n",
      "Installing collected packages: dlib, face-recognition\n",
      "  Running setup.py install for dlib: started\n",
      "  Running setup.py install for dlib: finished with status 'error'\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  error: subprocess-exited-with-error\n",
      "  \n",
      "  python setup.py bdist_wheel did not run successfully.\n",
      "  exit code: 1\n",
      "  \n",
      "  [70 lines of output]\n",
      "  running bdist_wheel\n",
      "  running build\n",
      "  running build_py\n",
      "  running build_ext\n",
      "  C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-43l2juk_\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\setup.py:129: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.\n",
      "    if LooseVersion(cmake_version) < '3.1.0':\n",
      "  Building extension for Python 3.9.13 (main, Aug 25 2022, 23:51:50) [MSC v.1916 64 bit (AMD64)]\n",
      "  Invoking CMake setup: 'cmake C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-43l2juk_\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\tools\\python -DCMAKE_LIBRARY_OUTPUT_DIRECTORY=C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-43l2juk_\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\build\\lib.win-amd64-cpython-39 -DPYTHON_EXECUTABLE=C:\\Users\\Sahil Raghuvanshi\\ana\\python.exe -DCMAKE_LIBRARY_OUTPUT_DIRECTORY_RELEASE=C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-43l2juk_\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\build\\lib.win-amd64-cpython-39 -A x64'\n",
      "  -- Building for: NMake Makefiles\n",
      "  CMake Error at CMakeLists.txt:5 (message):\n",
      "  \n",
      "  \n",
      "  \n",
      "    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
      "  \n",
      "  \n",
      "    You must use Visual Studio to build a python extension on windows.  If you\n",
      "    are getting this error it means you have not installed Visual C++.  Note\n",
      "    that there are many flavors of Visual Studio, like Visual Studio for C#\n",
      "    development.  You need to install Visual Studio for C++.\n",
      "  \n",
      "  \n",
      "    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
      "  \n",
      "  \n",
      "  \n",
      "  \n",
      "  -- Configuring incomplete, errors occurred!\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 2, in <module>\n",
      "    File \"<pip-setuptools-caller>\", line 34, in <module>\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-43l2juk_\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\setup.py\", line 222, in <module>\n",
      "      setup(\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\__init__.py\", line 87, in setup\n",
      "      return distutils.core.setup(**attrs)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\core.py\", line 185, in setup\n",
      "      return run_commands(dist)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\core.py\", line 201, in run_commands\n",
      "      dist.run_commands()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 973, in run_commands\n",
      "      self.run_command(cmd)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\dist.py\", line 1217, in run_command\n",
      "      super().run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 992, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\wheel\\bdist_wheel.py\", line 299, in run\n",
      "      self.run_command('build')\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\cmd.py\", line 319, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\dist.py\", line 1217, in run_command\n",
      "      super().run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 992, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\command\\build.py\", line 24, in run\n",
      "      super().run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\command\\build.py\", line 132, in run\n",
      "      self.run_command(cmd_name)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\cmd.py\", line 319, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\dist.py\", line 1217, in run_command\n",
      "      super().run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 992, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-43l2juk_\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\setup.py\", line 134, in run\n",
      "      self.build_extension(ext)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-43l2juk_\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\setup.py\", line 171, in build_extension\n",
      "      subprocess.check_call(cmake_setup, cwd=build_folder)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\subprocess.py\", line 373, in check_call\n",
      "      raise CalledProcessError(retcode, cmd)\n",
      "  subprocess.CalledProcessError: Command '['cmake', 'C:\\\\Users\\\\Sahil Raghuvanshi\\\\AppData\\\\Local\\\\Temp\\\\pip-install-43l2juk_\\\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\\\tools\\\\python', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=C:\\\\Users\\\\Sahil Raghuvanshi\\\\AppData\\\\Local\\\\Temp\\\\pip-install-43l2juk_\\\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\\\build\\\\lib.win-amd64-cpython-39', '-DPYTHON_EXECUTABLE=C:\\\\Users\\\\Sahil Raghuvanshi\\\\ana\\\\python.exe', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_RELEASE=C:\\\\Users\\\\Sahil Raghuvanshi\\\\AppData\\\\Local\\\\Temp\\\\pip-install-43l2juk_\\\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\\\build\\\\lib.win-amd64-cpython-39', '-A', 'x64']' returned non-zero exit status 1.\n",
      "  [end of output]\n",
      "  \n",
      "  note: This error originates from a subprocess, and is likely not a problem with pip.\n",
      "  ERROR: Failed building wheel for dlib\n",
      "  error: subprocess-exited-with-error\n",
      "  \n",
      "  Running setup.py install for dlib did not run successfully.\n",
      "  exit code: 1\n",
      "  \n",
      "  [74 lines of output]\n",
      "  running install\n",
      "  C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\command\\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.\n",
      "    warnings.warn(\n",
      "  running build\n",
      "  running build_py\n",
      "  running build_ext\n",
      "  C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-43l2juk_\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\setup.py:129: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.\n",
      "    if LooseVersion(cmake_version) < '3.1.0':\n",
      "  Building extension for Python 3.9.13 (main, Aug 25 2022, 23:51:50) [MSC v.1916 64 bit (AMD64)]\n",
      "  Invoking CMake setup: 'cmake C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-43l2juk_\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\tools\\python -DCMAKE_LIBRARY_OUTPUT_DIRECTORY=C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-43l2juk_\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\build\\lib.win-amd64-cpython-39 -DPYTHON_EXECUTABLE=C:\\Users\\Sahil Raghuvanshi\\ana\\python.exe -DCMAKE_LIBRARY_OUTPUT_DIRECTORY_RELEASE=C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-43l2juk_\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\build\\lib.win-amd64-cpython-39 -A x64'\n",
      "  -- Building for: NMake Makefiles\n",
      "  CMake Error at CMakeLists.txt:5 (message):\n",
      "  \n",
      "  \n",
      "  \n",
      "    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
      "  \n",
      "  \n",
      "    You must use Visual Studio to build a python extension on windows.  If you\n",
      "    are getting this error it means you have not installed Visual C++.  Note\n",
      "    that there are many flavors of Visual Studio, like Visual Studio for C#\n",
      "    development.  You need to install Visual Studio for C++.\n",
      "  \n",
      "  \n",
      "    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
      "  \n",
      "  \n",
      "  \n",
      "  \n",
      "  -- Configuring incomplete, errors occurred!\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 2, in <module>\n",
      "    File \"<pip-setuptools-caller>\", line 34, in <module>\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-43l2juk_\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\setup.py\", line 222, in <module>\n",
      "      setup(\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\__init__.py\", line 87, in setup\n",
      "      return distutils.core.setup(**attrs)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\core.py\", line 185, in setup\n",
      "      return run_commands(dist)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\core.py\", line 201, in run_commands\n",
      "      dist.run_commands()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 973, in run_commands\n",
      "      self.run_command(cmd)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\dist.py\", line 1217, in run_command\n",
      "      super().run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 992, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\command\\install.py\", line 68, in run\n",
      "      return orig.install.run(self)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\command\\install.py\", line 692, in run\n",
      "      self.run_command('build')\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\cmd.py\", line 319, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\dist.py\", line 1217, in run_command\n",
      "      super().run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 992, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\command\\build.py\", line 24, in run\n",
      "      super().run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\command\\build.py\", line 132, in run\n",
      "      self.run_command(cmd_name)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\cmd.py\", line 319, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\dist.py\", line 1217, in run_command\n",
      "      super().run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 992, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-43l2juk_\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\setup.py\", line 134, in run\n",
      "      self.build_extension(ext)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-43l2juk_\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\setup.py\", line 171, in build_extension\n",
      "      subprocess.check_call(cmake_setup, cwd=build_folder)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\subprocess.py\", line 373, in check_call\n",
      "      raise CalledProcessError(retcode, cmd)\n",
      "  subprocess.CalledProcessError: Command '['cmake', 'C:\\\\Users\\\\Sahil Raghuvanshi\\\\AppData\\\\Local\\\\Temp\\\\pip-install-43l2juk_\\\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\\\tools\\\\python', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=C:\\\\Users\\\\Sahil Raghuvanshi\\\\AppData\\\\Local\\\\Temp\\\\pip-install-43l2juk_\\\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\\\build\\\\lib.win-amd64-cpython-39', '-DPYTHON_EXECUTABLE=C:\\\\Users\\\\Sahil Raghuvanshi\\\\ana\\\\python.exe', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_RELEASE=C:\\\\Users\\\\Sahil Raghuvanshi\\\\AppData\\\\Local\\\\Temp\\\\pip-install-43l2juk_\\\\dlib_3d5ecc4f6e8843f19ae0e94e1d8d3e29\\\\build\\\\lib.win-amd64-cpython-39', '-A', 'x64']' returned non-zero exit status 1.\n",
      "  [end of output]\n",
      "  \n",
      "  note: This error originates from a subprocess, and is likely not a problem with pip.\n",
      "error: legacy-install-failure\n",
      "\n",
      "Encountered error while trying to install package.\n",
      "\n",
      "dlib\n",
      "\n",
      "note: This is an issue with the package mentioned above, not pip.\n",
      "hint: See above for output from the failure.\n"
     ]
    }
   ],
   "source": [
    "pip install face-recognition"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "cf1c9e10",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: numpy in c:\\users\\sahil raghuvanshi\\ana\\lib\\site-packages (1.21.5)Note: you may need to restart the kernel to use updated packages.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "pip install numpy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c1b0a4a8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting face-recognitionNote: you may need to restart the kernel to use updated packages.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  error: subprocess-exited-with-error\n",
      "  \n",
      "  python setup.py bdist_wheel did not run successfully.\n",
      "  exit code: 1\n",
      "  \n",
      "  [70 lines of output]\n",
      "  running bdist_wheel\n",
      "  running build\n",
      "  running build_py\n",
      "  running build_ext\n",
      "  C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-q3e546fp\\dlib_a36de85350b1493eaf51f7af5292e3b7\\setup.py:129: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.\n",
      "    if LooseVersion(cmake_version) < '3.1.0':\n",
      "  Building extension for Python 3.9.13 (main, Aug 25 2022, 23:51:50) [MSC v.1916 64 bit (AMD64)]\n",
      "  Invoking CMake setup: 'cmake C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-q3e546fp\\dlib_a36de85350b1493eaf51f7af5292e3b7\\tools\\python -DCMAKE_LIBRARY_OUTPUT_DIRECTORY=C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-q3e546fp\\dlib_a36de85350b1493eaf51f7af5292e3b7\\build\\lib.win-amd64-cpython-39 -DPYTHON_EXECUTABLE=C:\\Users\\Sahil Raghuvanshi\\ana\\python.exe -DCMAKE_LIBRARY_OUTPUT_DIRECTORY_RELEASE=C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-q3e546fp\\dlib_a36de85350b1493eaf51f7af5292e3b7\\build\\lib.win-amd64-cpython-39 -A x64'\n",
      "  -- Building for: NMake Makefiles\n",
      "  CMake Error at CMakeLists.txt:5 (message):\n",
      "  \n",
      "  \n",
      "  \n",
      "    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
      "  \n",
      "  \n",
      "    You must use Visual Studio to build a python extension on windows.  If you\n",
      "    are getting this error it means you have not installed Visual C++.  Note\n",
      "    that there are many flavors of Visual Studio, like Visual Studio for C#\n",
      "    development.  You need to install Visual Studio for C++.\n",
      "  \n",
      "  \n",
      "    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
      "  \n",
      "  \n",
      "  \n",
      "  \n",
      "  -- Configuring incomplete, errors occurred!\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 2, in <module>\n",
      "    File \"<pip-setuptools-caller>\", line 34, in <module>\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-q3e546fp\\dlib_a36de85350b1493eaf51f7af5292e3b7\\setup.py\", line 222, in <module>\n",
      "      setup(\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\__init__.py\", line 87, in setup\n",
      "      return distutils.core.setup(**attrs)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\core.py\", line 185, in setup\n",
      "      return run_commands(dist)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\core.py\", line 201, in run_commands\n",
      "      dist.run_commands()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 973, in run_commands\n",
      "      self.run_command(cmd)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\dist.py\", line 1217, in run_command\n",
      "      super().run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 992, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\wheel\\bdist_wheel.py\", line 299, in run\n",
      "      self.run_command('build')\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\cmd.py\", line 319, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\dist.py\", line 1217, in run_command\n",
      "      super().run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 992, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\command\\build.py\", line 24, in run\n",
      "      super().run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\command\\build.py\", line 132, in run\n",
      "      self.run_command(cmd_name)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\cmd.py\", line 319, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\dist.py\", line 1217, in run_command\n",
      "      super().run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 992, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-q3e546fp\\dlib_a36de85350b1493eaf51f7af5292e3b7\\setup.py\", line 134, in run\n",
      "      self.build_extension(ext)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-q3e546fp\\dlib_a36de85350b1493eaf51f7af5292e3b7\\setup.py\", line 171, in build_extension\n",
      "      subprocess.check_call(cmake_setup, cwd=build_folder)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\subprocess.py\", line 373, in check_call\n",
      "      raise CalledProcessError(retcode, cmd)\n",
      "  subprocess.CalledProcessError: Command '['cmake', 'C:\\\\Users\\\\Sahil Raghuvanshi\\\\AppData\\\\Local\\\\Temp\\\\pip-install-q3e546fp\\\\dlib_a36de85350b1493eaf51f7af5292e3b7\\\\tools\\\\python', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=C:\\\\Users\\\\Sahil Raghuvanshi\\\\AppData\\\\Local\\\\Temp\\\\pip-install-q3e546fp\\\\dlib_a36de85350b1493eaf51f7af5292e3b7\\\\build\\\\lib.win-amd64-cpython-39', '-DPYTHON_EXECUTABLE=C:\\\\Users\\\\Sahil Raghuvanshi\\\\ana\\\\python.exe', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_RELEASE=C:\\\\Users\\\\Sahil Raghuvanshi\\\\AppData\\\\Local\\\\Temp\\\\pip-install-q3e546fp\\\\dlib_a36de85350b1493eaf51f7af5292e3b7\\\\build\\\\lib.win-amd64-cpython-39', '-A', 'x64']' returned non-zero exit status 1.\n",
      "  [end of output]\n",
      "  \n",
      "  note: This error originates from a subprocess, and is likely not a problem with pip.\n",
      "  ERROR: Failed building wheel for dlib\n",
      "  error: subprocess-exited-with-error\n",
      "  \n",
      "  Running setup.py install for dlib did not run successfully.\n",
      "  exit code: 1\n",
      "  \n",
      "  [74 lines of output]\n",
      "  running install\n",
      "  C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\command\\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.\n",
      "    warnings.warn(\n",
      "  running build\n",
      "  running build_py\n",
      "  running build_ext\n",
      "  C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-q3e546fp\\dlib_a36de85350b1493eaf51f7af5292e3b7\\setup.py:129: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.\n",
      "    if LooseVersion(cmake_version) < '3.1.0':\n",
      "  Building extension for Python 3.9.13 (main, Aug 25 2022, 23:51:50) [MSC v.1916 64 bit (AMD64)]\n",
      "  Invoking CMake setup: 'cmake C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-q3e546fp\\dlib_a36de85350b1493eaf51f7af5292e3b7\\tools\\python -DCMAKE_LIBRARY_OUTPUT_DIRECTORY=C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-q3e546fp\\dlib_a36de85350b1493eaf51f7af5292e3b7\\build\\lib.win-amd64-cpython-39 -DPYTHON_EXECUTABLE=C:\\Users\\Sahil Raghuvanshi\\ana\\python.exe -DCMAKE_LIBRARY_OUTPUT_DIRECTORY_RELEASE=C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-q3e546fp\\dlib_a36de85350b1493eaf51f7af5292e3b7\\build\\lib.win-amd64-cpython-39 -A x64'\n",
      "  -- Building for: NMake Makefiles\n",
      "  CMake Error at CMakeLists.txt:5 (message):\n",
      "  \n",
      "  \n",
      "  \n",
      "    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
      "  \n",
      "  \n",
      "    You must use Visual Studio to build a python extension on windows.  If you\n",
      "    are getting this error it means you have not installed Visual C++.  Note\n",
      "    that there are many flavors of Visual Studio, like Visual Studio for C#\n",
      "    development.  You need to install Visual Studio for C++.\n",
      "  \n",
      "  \n",
      "    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
      "  \n",
      "  \n",
      "  \n",
      "  \n",
      "  -- Configuring incomplete, errors occurred!\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 2, in <module>\n",
      "    File \"<pip-setuptools-caller>\", line 34, in <module>\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-q3e546fp\\dlib_a36de85350b1493eaf51f7af5292e3b7\\setup.py\", line 222, in <module>\n",
      "      setup(\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\__init__.py\", line 87, in setup\n",
      "      return distutils.core.setup(**attrs)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\core.py\", line 185, in setup\n",
      "      return run_commands(dist)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\core.py\", line 201, in run_commands\n",
      "      dist.run_commands()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 973, in run_commands\n",
      "      self.run_command(cmd)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\dist.py\", line 1217, in run_command\n",
      "      super().run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 992, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\command\\install.py\", line 68, in run\n",
      "      return orig.install.run(self)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\command\\install.py\", line 692, in run\n",
      "      self.run_command('build')\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\cmd.py\", line 319, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\dist.py\", line 1217, in run_command\n",
      "      super().run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 992, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\command\\build.py\", line 24, in run\n",
      "      super().run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\command\\build.py\", line 132, in run\n",
      "      self.run_command(cmd_name)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\cmd.py\", line 319, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\dist.py\", line 1217, in run_command\n",
      "      super().run_command(command)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\site-packages\\setuptools\\_distutils\\dist.py\", line 992, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-q3e546fp\\dlib_a36de85350b1493eaf51f7af5292e3b7\\setup.py\", line 134, in run\n",
      "      self.build_extension(ext)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\AppData\\Local\\Temp\\pip-install-q3e546fp\\dlib_a36de85350b1493eaf51f7af5292e3b7\\setup.py\", line 171, in build_extension\n",
      "      subprocess.check_call(cmake_setup, cwd=build_folder)\n",
      "    File \"C:\\Users\\Sahil Raghuvanshi\\ana\\lib\\subprocess.py\", line 373, in check_call\n",
      "      raise CalledProcessError(retcode, cmd)\n",
      "  subprocess.CalledProcessError: Command '['cmake', 'C:\\\\Users\\\\Sahil Raghuvanshi\\\\AppData\\\\Local\\\\Temp\\\\pip-install-q3e546fp\\\\dlib_a36de85350b1493eaf51f7af5292e3b7\\\\tools\\\\python', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=C:\\\\Users\\\\Sahil Raghuvanshi\\\\AppData\\\\Local\\\\Temp\\\\pip-install-q3e546fp\\\\dlib_a36de85350b1493eaf51f7af5292e3b7\\\\build\\\\lib.win-amd64-cpython-39', '-DPYTHON_EXECUTABLE=C:\\\\Users\\\\Sahil Raghuvanshi\\\\ana\\\\python.exe', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_RELEASE=C:\\\\Users\\\\Sahil Raghuvanshi\\\\AppData\\\\Local\\\\Temp\\\\pip-install-q3e546fp\\\\dlib_a36de85350b1493eaf51f7af5292e3b7\\\\build\\\\lib.win-amd64-cpython-39', '-A', 'x64']' returned non-zero exit status 1.\n",
      "  [end of output]\n",
      "  \n",
      "  note: This error originates from a subprocess, and is likely not a problem with pip.\n",
      "error: legacy-install-failure\n",
      "\n",
      "Encountered error while trying to install package.\n",
      "\n",
      "dlib\n",
      "\n",
      "note: This is an issue with the package mentioned above, not pip.\n",
      "hint: See above for output from the failure.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  Using cached face_recognition-1.3.0-py2.py3-none-any.whl (15 kB)\n",
      "Requirement already satisfied: numpy in c:\\users\\sahil raghuvanshi\\ana\\lib\\site-packages (from face-recognition) (1.21.5)\n",
      "Requirement already satisfied: face-recognition-models>=0.3.0 in c:\\users\\sahil raghuvanshi\\ana\\lib\\site-packages (from face-recognition) (0.3.0)\n",
      "Requirement already satisfied: Click>=6.0 in c:\\users\\sahil raghuvanshi\\ana\\lib\\site-packages (from face-recognition) (8.0.4)\n",
      "Collecting dlib>=19.7\n",
      "  Using cached dlib-19.24.1.tar.gz (3.2 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Requirement already satisfied: Pillow in c:\\users\\sahil raghuvanshi\\ana\\lib\\site-packages (from face-recognition) (9.2.0)\n",
      "Requirement already satisfied: colorama in c:\\users\\sahil raghuvanshi\\ana\\lib\\site-packages (from Click>=6.0->face-recognition) (0.4.5)\n",
      "Building wheels for collected packages: dlib\n",
      "  Building wheel for dlib (setup.py): started\n",
      "  Building wheel for dlib (setup.py): finished with status 'error'\n",
      "  Running setup.py clean for dlib\n",
      "Failed to build dlib\n",
      "Installing collected packages: dlib, face-recognition\n",
      "  Running setup.py install for dlib: started\n",
      "  Running setup.py install for dlib: finished with status 'error'\n"
     ]
    }
   ],
   "source": [
    "pip install face-recognition"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7295e0e5",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
